packages/release-mastermind/src/contexts/projects.ts

```diff
@@ -8,98 +1,215 @@
+			throw err
+		})
+
+		const currentVersion: Version = await utils.versioning
+			.parse(config, config.project?.ref)
+			.catch((err) => {
+				log(LoggingLevels.error, `Error thrown while parsing versioning: `, err)
+				throw err
+			})
+
+		let localProject
+		localProject = await utils.api.project.projects.get(
+			project.project_url.split("/").pop()
+		)
+		const changes = context.payload.changes
+		const localColumn = await utils.api.project.column.get(project.column_id)
+
+		const localCard = await utils.api.project.card.get(project.id)
+
+		return {
+			sha: context.sha,
+			action: context.payload.action as string,
+			currentVersion,
+			IDNumber: issue.id,
+			props: {
+				type: "project",
+				ID: issue.number,
+				creator: issue.user.login,
+				description: issue.body || "",
+				locked: issue.locked,
+				state: issue.state as ProjectContext["props"]["state"],
+				title: issue.title,
+				project: localProject,
+				column_id: project.column_id,
+				localColumn,
+				localCard,
+				changes,
+				labels
+			}
+		}
+	}
+
+	async run(attempt?: number) {
+		if (!this.config)
+			throw new LoggingDataClass(
+				LoggingLevels.error,
+				"Cannot start without config"
+			)
+		if (!attempt) {
+			attempt = 1
+			core.startGroup("project Actions")
+		}
+		if (!attempt) {
+			attempt = 1
+			core.startGroup("project Actions")
+		}
+		const seconds = attempt * 10
+		let enforceConventionsSuccess = true
+
+		try {
+			if (this.config.enforceConventions && this.util.shouldRun("convention")) {
+				if (!this.config.enforceConventions.onColumn) return
+				this.config.enforceConventions.onColumn =
+					await this.convertColumnStringsToIDArray(
+						this.config.enforceConventions.onColumn
+					)
+				if (
+					this.config.enforceConventions?.onColumn?.includes(
+						this.context.props.column_id
+					)
+				)
+					enforceConventionsSuccess = await this.conventions.enforce(this)
+			}
 
-  async run(attempt?: number) {
-    if (!this.config)
-      throw new LoggingDataClass(LoggingLevels.error, 'Cannot start without config')
-    if (!attempt) {
-      attempt = 1
-      core.startGroup('project Actions')
-    }
-    if (!attempt) {
-      attempt = 1
-      core.startGroup('project Actions')
-    }
-    let seconds = attempt * 10,
-      enforceConventionsSuccess: boolean = true
+			if (enforceConventionsSuccess) {
+				// some code
+				if (this.config.labels && this.util.shouldRun("label"))
+					await this.applyLabels(this).catch((err) => {
+						log(LoggingLevels.error, "Error applying labels", err)
+					})
+				if (this.config.syncRemote && this.util.shouldRun("release"))
+					await this.syncRemoteProject(this).catch((err) => {
+						log(LoggingLevels.error, "Error syncing remote project", err)
+					})
+				core.endGroup()
+			}
+		} catch (err) {
+			if (attempt > this.retryLimit) {
+				core.endGroup()
+				throw log(
+					LoggingLevels.emergency,
+					`project actions failed. Terminating job.`
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Videndum/action-masterminds/pullRequest?prid=7872005)
```suggestion
          "project actions failed. Terminating job."
```